specification/dartLangSpec.tex

```diff
@@ -5282,26 +5221,44 @@ \subsubsection{Inheritance and Overriding}
 \end{itemize}
 
 \LMHash{}%
-Let $I$ be the interface of a class $C$ declared in library $L$.
-$I$ \Index{inherits} all members of $\inherited{I, L}$
-and $I$ \Index{overrides} $m'$ if $m' \in \overrides{I, L}$.
+Let $I$ be the interface of a class $C$ declared in a library $L$,
+and let $K$ be a library.
+We say that the set of members that $I$
+\IndexCustom{inherits with respect to $K$}{interface!inherits}
+is $\inherited{I, K}$,
```

Comment:
Yes, `$m \in \inherited{I, K}$` is the same thing as '$m$ is inherited by $i$ with respect to $K$'. We could use the natural language phrase in the definition, but function definitions seem to be more naturally expressed in terms of function notation. Conversely, a natural language phrase might seem more natural in natural language text in the rest of the spec.

We do use function notation in a number of newer parts of the spec, e.g., `\NonnullTypeof{T}` rather than "the non-nullable type corresponding to $T$`, so we could change this as well. It's actually only used in a handful of locations.

Reply:
I left the occurrences of 'inherits'/'inherited' in the spec unchanged for now. We use the concept both when talking about interfaces (where it includes members of all superinterfaces, but only considers the direct superinterfaces because they in turn hold all members that are inherited indirectly from an indirect superinterface) and when talking about implementations of members with classes (so it's only about the chain of superclasses, but it considers the whole chain, because we don't want to say that any intermediate, possibly abstract, classes will also inherit an implementation whenever one is available).

So we might want to change this such that we use two different words for the "inheritance" of member signatures among interfaces, vs. the "inheritance" of member implementations among elements in the superclass chain.